plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/condition/BrowserContainsText.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat, Inc and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.reddeer.eclipse.condition;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.reddeer.common.condition.AbstractWaitCondition;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.swt.condition.PageIsLoaded;
+import org.eclipse.reddeer.swt.impl.browser.InternalBrowser;
+
+/**
+ * Wait condition which returns true if a given browser has the specified URL and specific text.
+ */
+public class BrowserContainsText extends AbstractWaitCondition {
+
+	private InternalBrowser internalBrowser;
+	private String text;
+	private String url;
+	private boolean refresh;
+	
+	public BrowserContainsText(InternalBrowser browser, String url, String text, boolean refresh) {
+		this.internalBrowser = browser;
+		this.url = url;
+		this.text = text;
+		this.refresh = refresh;
+	}
+	
+	public BrowserContainsText(InternalBrowser browser, String url, String text) {
+		this(browser, url, text, true);
+	}
+
+	public BrowserContainsText(String url, String text) {
+		this(new InternalBrowser(), url, text, true);
+	}
+	
+	public BrowserContainsText(String text, boolean refresh) {
+		this(new InternalBrowser(), null, text, refresh);
+	}
+
+	public BrowserContainsText(String text) {
+		this(new InternalBrowser(), null, text, true);
+	}
+
+	@Override
+	public boolean test() {
+		internalBrowser = new InternalBrowser();
+		if (url != null) {
+			internalBrowser.setURL(url);
```

Comment:
seems wrong to set URL everytime a condition is being tested. It should be done only once or it should be left to user. In the second case we should not have a constructor with url parameter.